### PR TITLE
Supported CBLAS for fully_connected_op. 

### PR DESCRIPTION
**Pull request by jimbozhang**
Add a new fully connected op: fully_connected_op_cblas, which uses CBLAS as its back-end.

### CODE REVIEW

File: tiny_dnn/core/kernels/fully_connected_op_cblas.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  memcpy(output, bias.data(), sizeof(float_t) * out_size);
+#ifdef CNN_USE_DOUBLE
+  cblas_dgemm(CblasRowMajor, CblasNoTrans, CblasNoTrans, 1, out_size, in_size,
+              alpha, input, in_size, weight, out_size, beta, output, out_size);
+#else
+  cblas_sgemm(CblasRowMajor, CblasNoTrans, CblasNoTrans, 1, out_size, in_size,
+              alpha, input, in_size, weight, out_size, beta, output, out_size);
+#endif
+
+#endif  // CNN_USE_CBLAS
```
Comment (beru) [inline tiny_dnn/core/kernels/fully_connected_op_cblas.h]: if `CNN_USE_CBLAS` isn't defined but `fully_connected_op_cblas` is called, throwing `nn_error` exception might be preferrable. But... how to deal with the situation isn't consistent throughout the library. So I guess it's not mandatory.  :confused:

By the way, could you kindly tell me which library I should use to test this layer? OpenBLAS?

**Comment by beru** [comment]
I'm not so sure why travis-ci jobs failed. Checking now...

**Comment by beru** [comment]
travis-ci osx jobs fail.

https://travis-ci.org/tiny-dnn/tiny-dnn/jobs/340090985#L111

```
++brew tap
++grep homebrew/science
Updating Homebrew...
+'[' -z ']'
+brew tap homebrew/science
Error: homebrew/science was deprecated. This tap is now empty as all its formulae were migrated.
```

I guess below file needs to be updated.
https://github.com/tiny-dnn/tiny-dnn/blob/cfedbbe155e90c8da3553c045205ca30dc827974/.travis/install.sh

**Comment by beru** [comment]
@jimbozhang Once #931 is merged, travis-ci osx builds should pass.

**Comment by edgarriba** [comment]
@jimbozhang please update the readme with a quick description about how to setup this backend

**Comment by jimbozhang** [comment]
@edgarriba OK, I'll update the readme soon.

**Comment by jimbozhang** [comment]
@edgarriba please check the PR #935, where I added an example to illustrate how to setup it.

**Pull request merged**